src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke.util;
+
+import duke.Duke;
+import duke.exception.EmptyDescriptionException;
+import duke.exception.InvalidCommandException;
+
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+import java.io.IOException;
+
+/**
+ * Represents a Parser that parses user input for SeeWhyAre bot to perform corresponding actions.
+ *
+ * <p>CS2103T AY23/24 Semester 1
+ * Individual Project
+ * SeeWhyAre Bot
+ * 31 Aug 2023
+ *
+ * @author Freddy Chen You Ren
+ */
+public class Parser {
+    private static String HORIZONTAL_LINE = "    ____________________________________________________________"; //60 underscores.
+    private TaskList taskList;
+    private Ui ui;
+    public enum Command {
+        BYE, LIST, MARK, UNMARK, DELETE, FIND, DEADLINE, TODO, EVENT, UNKNOWN
+    }
+
+    /**
+     * Constructs a Parser with the specified task list and CLI user interface.
+     *
+     * @param taskList The task list to work with.
+     * @param ui The CLI user interface to display messages.
+     */
+    public Parser(TaskList taskList, Ui ui) {
+        this.taskList = taskList;
+        this.ui = ui;
+    }
+
+    private void printHorizontalLine() {
+        System.out.println(HORIZONTAL_LINE);
+    }
+
+    /**
+     * Retrieves the command keyword from the user input.
+     * The command keyword is the first word from the user input.
+     *
+     * @param line The user input to be parsed.
+     * @return The command keyword extracted from the user input.
+     */
+    private String getCommand(String line) {
+        return line.trim().split("\\s+")[0].toUpperCase();
+    }
+
+    /**
+     * Parses user input and performs the corresponding action.
+     *
+     * @param userInput The user's input to be parsed.
+     * @throws IOException If there is an issue with I/O operations.
+     * @throws EmptyDescriptionException If a description is required but not provided.
+     * @throws InvalidCommandException If an invalid command is entered.
+     */
+    public void parseInput(String userInput)
+            throws IOException, EmptyDescriptionException, InvalidCommandException {
+        try {
+            String firstWord = getCommand(userInput);
+            String[] words = userInput.split("\\s+"); // Split input by space, put into array
+            Command command; //Use enum
+            try {
+                command = Command.valueOf(firstWord);
+            } catch (IllegalArgumentException e) {
+                command = Command.UNKNOWN;
```

Comment:
You can immediately throw the `InvalidCommandException` here and not need an `UNKNOWN` enum